datafusion/core/tests/sqllogictests/test_files/scalar.slt

```diff
@@ -911,17 +911,15 @@ SELECT not(true), not(false)
 ----
 false true
 
-query BB
+query error DataFusion error: Optimizer rule 'simplify_expressions' failed\ncaused by\nInternal error: NOT 'Literal \{ value: Int64\(1\) \}' can't be evaluated because the expression's type is Int64, not boolean or NULL\. This was likely caused by a bug in DataFusion's code and we would welcome that you file an bug report in our issue tracker
```

Comment:
It should fail in `type coercion`, it's a future ticket.
We should try to cast param. (if fail cast then throw a error)